libraries/tools/kotlin-test-js-runner/karma-kotlin-reporter.js

```diff
@@ -124,6 +124,13 @@ const hashString = function (s) {
     return hash
 }
 
+const withoutSurroundingApostrophes = function (s) {
+    if (s.startsWith("'") && s.endsWith("'"))
```

Comment:
I constantly feel we could remove this `slice` at all. Could you check if some tests will fail?
Anyway, I approve the MR, and after that, I will think about removing this `slice` call.
Great job!

Reply:
When we don't read the output through the TeamCity adapter, every `String` that is passed in the `log` parameter is wrapped in apostrophes, while other types are not.

E.g.: 
```
println("hello") -> 'hello'
println(null)    -> null
```
This was the source of the original bug, because in 1.9.20 we didn't pass the console logs through the TeamCity adapter if I'm not mistaken, and `slice()` was called on `null`, so it became `ul`. At least the format of the output didn't look like it's coming from this adapter.

In the latest version it seems like the output gets through the TC adapter by default, which always passes them on as a `String`.

```
teamCity.sendMessage(
    messageType,
    withName(
        name,
        {
            "out": `[${type}] ${format(message, ...optionalParams)}\n`
        }
    )
)
```
As a result, what the karma reporter receives is always wrapped in apostrophes, so technically you always need to `slice()`. I only made the apostrophe stripping smarter to make the previous (1.9.20) case work too. As such, if we remove `slice()`, this test (and possibly every other with `println()`) will start failing.

```
expected:<...system-out><![CDATA[[[log] ====

[log] null
[log]

[log] ====]]]></system-out>
  ...> but was:<...system-out><![CDATA[['[log] ====

''[log] null
''[log]

''[log] ====

']]]></system-out>
```